sdk/search/azure-search/src/main/java/com/azure/search/models/SearchOptions.java

```diff
@@ -0,0 +1,562 @@
+     *
+     * @return the orderBy value.
+     */
+    public List<String> getOrderBy() {
+        return this.orderBy;
+    }
+
+    /**
+     * Set the orderBy property: The list of OData $orderby expressions by
+     * which to sort the results. Each expression can be either a field name or
+     * a call to either the geo.distance() or the search.score() functions.
+     * Each expression can be followed by asc to indicate ascending, and desc
+     * to indicate descending. The default is ascending order. Ties will be
+     * broken by the match scores of documents. If no OrderBy is specified, the
+     * default sort order is descending by document match score. There can be
+     * at most 32 $orderby clauses.
+     *
+     * @param orderBy the orderBy value to set.
+     * @return the SearchOptions object itself.
+     */
+    public SearchOptions setOrderBy(String... orderBy) {
+        this.orderBy = Arrays.asList(orderBy);
+        return this;
+    }
+
+    /**
+     * Get the queryType property: A value that specifies the syntax of the
+     * search query. The default is 'simple'. Use 'full' if your query uses the
+     * Lucene query syntax. Possible values include: 'simple', 'full'.
+     *
+     * @return the queryType value.
+     */
+    public QueryType getQueryType() {
+        return this.queryType;
+    }
+
+    /**
+     * Set the queryType property: A value that specifies the syntax of the
+     * search query. The default is 'simple'. Use 'full' if your query uses the
+     * Lucene query syntax. Possible values include: 'simple', 'full'.
+     *
+     * @param queryType the queryType value to set.
+     * @return the SearchOptions object itself.
+     */
+    public SearchOptions setQueryType(QueryType queryType) {
+        this.queryType = queryType;
+        return this;
+    }
+
+    /**
+     * Get the scoringParameters property: The list of parameter values to be
+     * used in scoring functions (for example, referencePointParameter) using
+     * the format name-values. For example, if the scoring profile defines a
+     * function with a parameter called 'mylocation' the parameter string would
+     * be "mylocation--122.2,44.8" (without the quotes).
+     *
+     * @return the scoringParameters value.
+     */
+    public List<String> getScoringParameters() {
+        return this.scoringParameters;
+    }
+
+    /**
+     * Set the scoringParameters property: The list of parameter values to be
+     * used in scoring functions (for example, referencePointParameter) using
+     * the format name-values. For example, if the scoring profile defines a
+     * function with a parameter called 'mylocation' the parameter string would
+     * be "mylocation--122.2,44.8" (without the quotes).
+     *
+     * @param scoringParameters the scoringParameters value to set.
+     * @return the SearchOptions object itself.
+     */
+    public SearchOptions setScoringParameters(String... scoringParameters) {
```

Comment:
The .NET SDK has a very handy class to represent scoring parameters:

https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/search/Microsoft.Azure.Search.Data/src/Customizations/Documents/Models/ScoringParameter.cs

It helps customers deal with the otherwise baroque syntax, but it required some work to inject it into the generated code.

Since it would be a breaking change to do this later, we should figure out if we can do this for preview.

Reply:
I suggest doing this in a seperate PR after this one is merged, since it will require re-generating the code